dataplane/standalone/apigen/xml.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// Doxygen is the root of the generated xml struct.
+type Doxygen struct {
+	CompoundDef CompoundDef `xml:"compounddef"`
+}
+
+// CompoundDef contains a list sections in the xml.
+type CompoundDef struct {
+	Title      string       `xml:"title"`
+	SectionDef []SectionDef `xml:"sectiondef"`
+}
+
+// SectionDef contains a list of members.
+type SectionDef struct {
+	MemberDef []MemberDef `xml:"memberdef"`
+}
+
+// MemberDef is the definition of a single type.
+type MemberDef struct {
+	Name       string      `xml:"name"`
+	EnumValues []EnumValue `xml:"enumvalue"`
+	Kind       string      `xml:"kind,attr"`
+}
+
+// EnumValue is a single values in a enum.
+type EnumValue struct {
+	Name                string              `xml:"name"`
+	Initializer         string              `xml:"initializer"`
+	DetailedDescription DetailedDescription `xml:"detaileddescription"`
+}
+
+// DetailedDescription contains extra information about an enum value.
+type DetailedDescription struct {
+	Para Para `xml:"para"`
```

Comment:
I prefer Paragraph since "para" is a shortform that could refer to different things.